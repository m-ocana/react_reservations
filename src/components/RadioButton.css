
.radio_button {
  display: block;
  padding: 0 2px;
}

.label {
  display: inline-block;
}
