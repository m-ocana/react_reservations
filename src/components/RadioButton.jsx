import React, { Component, PropTypes } from 'react'
import style from './RadioButton.css'

export default class RadioButton extends Component {

  static PropTypes = {
    checked: PropTypes.bool,
    className: PropTypes.string,
    clickHandler: PropTypes.func,
    label: PropTypes.string,
    name: PropTypes.string,
    value: PropTypes.string
  }

  static defaultProps = {
    checked: false,
    className: '',
    clickHandler: () => {},
    label: '',
    name: '',
    value: ''
  }

  handleChange = (e) => {
    this.props.clickHandler(e)
  }

  render() {
    const { checked, className, label, name, value } = this.props
    return (
      <label
        className={style.radio_button}
        htmlFor={label}
      >
        <input
          checked={checked}
          className={style[`${className}`]}
          name={name}
          onChange={(e) => {
            if (e.target.checked !== checked) this.handleChange(e)
          }}
          type="radio"
          value={value}
        />
        { label &&
          <span className={style.label}>{label}</span>
        }
      </label>
    )
  }
}
