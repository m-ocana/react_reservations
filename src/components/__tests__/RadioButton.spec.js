import renderComponent from '../../helpers/test-component'
import RadioButton from '../RadioButton'

const renderRadioButton = renderComponent(RadioButton)
const initialProps = {
  className: 'radio-button',
  label: 'Yes',
  name: 'selection',
  value: 'on',
  clickHandler: jest.fn()
}

describe('<RadioButton /> component', () => {
  describe('@renders', () => {
    it('should render dumb RadioButton', () => {
      expect(renderRadioButton().getTree())
        .toMatchSnapshot()
    })

    it('should render RadioButton with initial props', () => {
      expect(renderRadioButton(initialProps).getTree())
        .toMatchSnapshot()
    })

    it('should render checked RadioButton', () => {
      expect(renderRadioButton({
        ...initialProps,
        checked: true
      }).getTree()).toMatchSnapshot()
    })
  })

  describe('@events', () => {
    beforeEach(() => jest.resetAllMocks())

    describe('onChange', () => {
      it('should trigger clickHandler', () => {
        const { wrapper, instance } = renderRadioButton(initialProps)
        wrapper.find('.radio-button').simulate('change', { target: { checked: true } })
        expect(instance.props.clickHandler).toHaveBeenCalled()
      })

      it('should not trigger clickHandler', () => {
        const { wrapper, instance } = renderRadioButton(initialProps)
        wrapper.find('.radio-button').simulate('change', { target: { checked: false } })
        expect(instance.props.clickHandler).not.toHaveBeenCalled()
      })

      it('should invoke handleChange function', () => {
        const { wrapper, instance } = renderRadioButton(initialProps)
        instance.handleChange = jest.fn()
        wrapper.find('.radio-button').simulate('change', { target: { checked: true } })
        expect(instance.handleChange).toHaveBeenCalled()
      })
    })
  })
})
