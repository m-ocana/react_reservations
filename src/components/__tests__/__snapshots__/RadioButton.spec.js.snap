// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RadioButton /> component @renders should render RadioButton with initial props 1`] = `
<label
  className="radio_button"
  htmlFor="Yes"
>
  <input
    checked={false}
    className="radio-button"
    name="selection"
    onChange={[Function]}
    type="radio"
    value="on"
  />
  <span
    className="label"
  >
    Yes
  </span>
</label>
`;

exports[`<RadioButton /> component @renders should render checked RadioButton 1`] = `
<label
  className="radio_button"
  htmlFor="Yes"
>
  <input
    checked={true}
    className="radio-button"
    name="selection"
    onChange={[Function]}
    type="radio"
    value="on"
  />
  <span
    className="label"
  >
    Yes
  </span>
</label>
`;

exports[`<RadioButton /> component @renders should render dumb RadioButton 1`] = `
<label
  className="radio_button"
  htmlFor=""
>
  <input
    checked={false}
    className=""
    name=""
    onChange={[Function]}
    type="radio"
    value=""
  />
</label>
`;
